cost/google/cud_recommendations/google_committed_use_discount_recommendations.pt

```diff
@@ -220,65 +239,129 @@ script "js_instance_cost_mapping", type:"javascript" do
+      }
+      total += monthly_savings
+
+      var algorithm = ""
+      if (algorithm_mapping[param_algorithm]) {
+        algorithm = algorithm_mapping[param_algorithm].name
       }
-      strNano = "."+recommendation.costNanos
-      combine = recommendation.costUnits + parseFloat(strNano)
-      monthsDuration = parseFloat(recommendation.duration.replace("s",""))/2628288
-      monthlySavings = combine / monthsDuration
-      total = total+monthlySavings
+
       _.each(recommendation.content.operationGroups, function(oGroup) {
         if(oGroup.operations != null) {
           _.each(oGroup.operations, function(operation) {
             if (operation.action == "add") {
+              var plan = ""
+              if (operation.value.plan == "TWELVE_MONTH"){
+                plan = "1 Year"
+              } else if (operation.value.plan == "THIRTY_SIX_MONTH") {
+                plan = "3 Year"
+              }
+              
               recommendation["resourceName"] = operation.value.name
-              recommendation["plan"] = operation.value.plan
+              recommendation["plan"] = plan
               recommendation["resourceType"] = operation.value.type
             }
           })
         }
       })
-      recommendation['savings']= (Math.round( monthlySavings * 1000) / 1000)
+
+      recommendation['savings'] = (Math.round( monthly_savings * 1000) / 1000)
+      console.log("Rec Savings:", recommendation.savings)
       recommendation['savingsCurrency'] = cur
       recommendation['service'] = "Storage"
+      recommendation['algorithm'] = algorithm
       tags = []
       recommendation['tags'] = tags
+      console.log("Recommendation:", recommendation)
       instances.push(recommendation)
     })
-    if (instances.length != 0){
-      if(count){
-        total=cur + ' '+formatNumber((Math.round(total * 100) / 100));
-        message="The total estimated monthly savings are " +total;
+
+    var instances = _.filter(instances, function(inst) {
+      return inst.plan == param_term
+    })
+
+
+    if (instances.length != 0) {
+      if (count) {
+        total=cur + ' ' + formatNumber((Math.round(total * 100) / 100));
+        message = "The total estimated monthly savings are " + total;
       } else {
-        message="The Flexera Optima system does not have any data to calculate savings for these resources";
+        message = "The Flexera Optima system does not have any data to calculate savings for these resources";
       }
-    }else{
-      message="unable to find resources between recommender and resource api";
+    } else {
+      message = "unable to find resources between recommender and resource api";
     }
-    result={
+    result = {
       "instances": instances,
       "message": message
     };
     result.instances = _.sortBy(result.instances,"region");
     result.instances = _.sortBy(result.instances,"projectNumber");
+    console.log(result.instances)
```

Comment:
Remove this `console.log`.

Reply:
Thanks @juanestrahcl , this was addressed.